projects/gloo/api/v1/upstream.proto

```
@@ -161,6 +163,14 @@ message Upstream {
     // (bool) If set to true, Envoy will ignore the health value of a host when processing its removal from service discovery.
     // This means that if active health checking is used, Envoy will not wait for the endpoint to go unhealthy before removing it.
     google.protobuf.BoolValue ignore_health_on_host_removal = 22;
+
+    // (bool) If set to true, Service Discovery update period will be triggered once the TTL is expired.
+	// If minimum TTL of all records is 0 then dns_refresh_rate will be used.
+	google.protobuf.BoolValue respect_dns_ttl= 29;
+
+    // (Duration) Service Discovery DNS Refresh Rate, if unset it defaults to 5 seconds.
+	// Minimum value is 1 ms
```

I like this callout. Is this lower bound something that is enforced by Envoy? Should we have logic in the control plane to validate that the provided duration is above the lower bound?

---

If envoy is enforcing the bounds, what is the behavior if a user sets a value below that lower bound? I think when there are requirements that Envoy has, we as the control-plane should perform that validation in order to reject resources which Envoy will reject

---

I think we want to avoid situations where Envoy is silently ignoring invalid configuration. One of the goals of the control plane is to provide feedback to users, so I think we should validate this and add an error to the report if the provided duration is outside the range that Envoy can support. @ianmacclancy what do you think?

---

Did we check for nacks? Envoy doesn't apply the config but it will update with new config or does it nack it?

---

I agree that we should add validation on dnsRefreshRate less than 1ms.

While not mentioned yet - I don't think we need to validate if the cluster type is STRICT_DNS or LOGICAL_DNS. If the cluster is not one of those types dnsRefreshRate is ignored but if we could ignore the error in cases where the cluster type is not strict/logical DNS that would be a nice add to the error checking.